include/alpaka/dev/DevOacc.hpp

```diff
@@ -306,6 +326,36 @@ namespace alpaka
                 generic::currentThreadWaitForDevice(dev);
             }
         };
+
+        //! The OpenACC platform device type trait specialization.
+        template<>
+        struct DevType<PltfOacc>
+        {
+            using type = DevOacc;
+        };
+
+        //! The OpenACC platform device get trait specialization.
+        template<>
+        struct GetDevByIdx<PltfOacc>
+        {
+            //! \param devIdx device id, less than GetDevCount
+            ALPAKA_FN_HOST static auto getDevByIdx(std::size_t devIdx) -> DevOacc
+            {
+                ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+                std::size_t const devCount(getDevCount<PltfOacc>());
```

Comment:
Why not make this `static`? Can the device count change during runtime?

Reply:
1. [Other](https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/pltf/PltfUniformCudaHipRt.hpp#L84) backends do not do this either. Although it sounds like a good idea.
2. As far as alpaka is concerned it should not change. OpenACC has a concept of a device type though. The [current implementation](https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/pltf/PltfOacc.hpp#L51) just returns the currently set device type, as Alpaka does not have any concept for this. In theory, if one would set a different device type between calls to this function, the returned count could change.

Reply:
1. [Other](https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/pltf/PltfUniformCudaHipRt.hpp#L84) backends do not do this either. Although it sounds like a good idea.
2. As far as alpaka is concerned it should not change. OpenACC has a concept of a device type though. The [current implementation](https://github.com/alpaka-group/alpaka/blob/develop/include/alpaka/pltf/PltfOacc.hpp#L51) just returns the currently set device type, as Alpaka does not have any concept for this. In theory, if one would set a different device type between calls to this function, the returned count could change.

Reply:
Btw.: USB-C GPU cases are a thing. These should be hot-plugable. What happens to, e.g., `cudaGetDeviceCount()` if you plug one in or remove one while the cuda runtime is initialized?